hnn_core/optimization/general_optimization.py

```diff
@@ -0,0 +1,408 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from .metrics import _rmse_evoked
+
+
+class Optimizer:
+    def __init__(self, net, tstop, constraints, set_params, solver='cobyla',
+                 obj_fun='dipole_rmse', scale_factor=1.,
+                 smooth_window_len=None, max_iter=200):
+        """Parameter optimization.
+
+        Parameters
+        ----------
+        net : Network
```

Comment:
```suggestion
        net : instance of Network
```

Reply:
I've always been curious about this: why do we say `net : instance of Network` but not `tstop : instance of float` even though both represent instances of Python objects.

Reply:
well `float` is not the class though, it's `Network` that is a class. Strictly speaking, `Network` is not a datatype even if you get `Network` if you do `type(Network)` ... instead it's more correct to say that `net` is an object that is an instance of the `Network` class.